airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -562,11 +577,38 @@ def extract_xcom(self, pod: V1Pod) -> str:
                 resp,
                 f"if [ -s {PodDefaults.XCOM_MOUNT_PATH}/return.json ]; then cat {PodDefaults.XCOM_MOUNT_PATH}/return.json; else echo __airflow_xcom_result_empty__; fi",  # noqa
             )
-            self._exec_pod_command(resp, "kill -s SIGINT 1")
+            if isinstance(result, str) and result.rstrip() != "__airflow_xcom_result_empty__":
```

Comment:
Is there a reason that you check for being a string? Can the K8s API return anything else than `Optional[str]`?
```suggestion
            if result and result.rstrip() != "__airflow_xcom_result_empty__":
```

Reply:
...or you could revert the order, check for `None`as below first, otherwise try to `json.loads()` afterwards?

Reply:
Static check was failing without checking for string. But made change you suggested - lets see if it passes

Reply:
Looks like it worked - thx @jens-scheffler-bosch